app/models/user.js

```diff
@@ -1,18 +1,36 @@
 'use strict';
 
+const bcrypt = require('../../node_modules/bcryptjs'),
+  logger = require('../logger'),
+  errors = require('../errors');
+
+const saltRounds = 10;
+
 module.exports = (sequelize, DataTypes) => {
-  const User = sequelize.define(
-    'User',
-    {
-      firstName: { type: DataTypes.STRING, allowNull: false },
-      lastName: { type: DataTypes.STRING, allowNull: false },
-      email: { type: DataTypes.STRING, allowNull: false, unique: true },
-      password: { type: DataTypes.STRING, allowNull: false }
-    },
-    {}
-  );
+  const User = sequelize.define('User', {
+    firstName: { type: DataTypes.STRING, allowNull: false },
+    lastName: { type: DataTypes.STRING, allowNull: false },
+    email: { type: DataTypes.STRING, allowNull: false, unique: true },
+    password: { type: DataTypes.STRING, allowNull: false }
+  });
   User.associate = function(models) {
     // associations can be defined here
   };
+
+  User.beforeCreate((user, options) => {
+    return bcrypt
+      .hash(user.password, saltRounds)
+      .then(hash => {
+        user.password = hash;
+      })
+      .catch(error => {
+        throw errors.defaultError(error.message);
+      });
+  });
+
+  User.prototype.printAfterCreationMessage = function printAfterCreationMessage() {
```

Comment:
dont overwrite object prototypes

Reply:
Same doubt as before

Reply:
its generally a bad practice. People cant really know .the changes a prototype may have suffered or someone may even change a prototype function that the library uses internally and mess things up without knowing so it is not recommended.